optuna/trial.py

```diff
@@ -393,7 +393,8 @@ def report(self, value, step=None):
 
         self.storage.set_trial_value(self._trial_id, value)
         if step is not None:
-            self.storage.set_trial_intermediate_value(self._trial_id, step, value)
+            if not self.storage.set_trial_intermediate_value(self._trial_id, step, value):
+                self.logger.warning('test msg')
```

Comment:
As I wrote in https://github.com/optuna/optuna/issues/852#issuecomment-589442838, I think that it's better to use [`warnings.warn()`](https://docs.python.org/3/library/warnings.html#warnings.warn) instead of `self.logger.warning()` to suppress duplicate log messages.